system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -317,24 +327,30 @@ class CLdapConfig : implements ILdapConfig, public CInterface
         else
             m_ldap_secure_port = atoi(portbuf.str());
 
+        int rc;
         StringBuffer hostbuf, dcbuf;
-        int rc = 0;
-        getLdapHost(hostbuf);
-        for(int retries = 0; retries <= LDAPSEC_MAX_RETRIES; retries++)
+        for (int numHosts=0; numHosts < getHostCount(); numHosts++)
         {
-            rc = LdapUtils::getServerInfo(hostbuf.str(), m_ldapport, dcbuf, m_serverType, cfg->queryProp(".//@ldapDomain"));
-            if(!LdapServerDown(rc) || retries > LDAPSEC_MAX_RETRIES)
-                break;
-            sleep(LDAPSEC_RETRY_WAIT);
-            if(retries < LDAPSEC_MAX_RETRIES)
+            getLdapHost(hostbuf);
+            for(int retries = 0; retries <= LDAPSEC_MAX_RETRIES; retries++)
             {
-                DBGLOG("Server %s temporarily unreachable.", hostbuf.str());
-                // Retrying next ldap sever, might be the same server
-                hostbuf.clear();
-                getLdapHost(hostbuf);
-                DBGLOG("Retrying with %s...", hostbuf.str());
+                rc = LdapUtils::getServerInfo(hostbuf.str(), m_ldapport, dcbuf, m_serverType, cfg->queryProp(".//@ldapDomain"));
```

Comment:
is it worthwhile extracting cfg->queryProp(".//@ldapDomain") once and storing in local var?